lib/services/access_request.go

```diff
@@ -1487,7 +1481,7 @@ func (m *RequestValidator) maxDurationForRole(roleName string) time.Duration {
 // approval). request TTL is capped to the smaller value between the const requestTTL and the
 // access request access expiry.
 func (m *RequestValidator) calculatePendingRequestTTL(r types.AccessRequest, now time.Time) (time.Duration, error) {
-	accessExpiryTTL := r.GetAccessExpiry().Sub(now)
+	accessExpiryTTL := r.GetMaxDuration().Sub(now)
```

Comment:
I'm not sure I fully understand the fix. E.g. why are we replacing access expiry with max duration everywhere?

It seems you're trying to address the issue on validation side? Shouldn't we be setting correct expiration on the created access request instead?

We should be really careful making changes here without understanding the full impact so we don't break existing access request guarantees/properties.